web/src/lib/components/memory-page/memory-viewer.svelte

```diff
@@ -90,32 +90,37 @@
 		}
 	};
 
-	let autoPlayInterval: NodeJS.Timeout;
 	let autoPlay = false;
 	let autoPlaySpeed = 5000;
 	let autoPlayProgress = 0;
 	let autoPlayIndex = 0;
-	let canPlayNext = true;
+	let autoPlayAnimationFrameRequest: number;
 
 	const toggleAutoPlay = () => {
```

Comment:
Maybe separate this logic into two functions `enableAutoPlay` and `DisableAutoPlay` so that it can be explicitly set from other functions autoPlayTransition